Sources/KeyboardShortcuts/SwiftUI+Extensions.swift

```diff
@@ -0,0 +1,123 @@
+// Created by Neil Clayton on 17/07/2024.
+// Using ideas by mbenoukaiss, https://github.com/sindresorhus/KeyboardShortcuts/issues/101
+
+
+import Foundation
+import SwiftUI
+import KeyboardShortcuts
+import Carbon
+
+// Provides a SwiftUI like wrapper func, that feels the same as the normal SwiftUI keyboardShortcut
+@available(macOS 11.0, *)
+extension View {
+	@ViewBuilder
+	public func keyboardShortcut(_ shortcutName: KeyboardShortcuts.Name) -> some View {
+		KeyboardShortcutView(shortcutName: shortcutName) {
+			self
+		}
+	}
+}
+
+// Holds the state of the shortcut, and changes that state when the shortcut changes
+// This causes the related NSMenuItem to also update (yipeee)
+@available(macOS 11.0, *)
+struct KeyboardShortcutView<Content: View>: View {
+	@State private var shortcutName: KeyboardShortcuts.Name
+	@State private var shortcut: KeyboardShortcuts.Shortcut?
+
+	private var content: () -> Content
+
+	init(shortcutName: KeyboardShortcuts.Name, content: @escaping () -> Content) {
+		self.shortcutName = shortcutName
+		self.shortcut = KeyboardShortcuts.getShortcut(for: shortcutName)
+		self.content = content
+	}
+
+	@ViewBuilder
+	var shortcutBody: some View {
+		if let shortcut, let keyEquivalent = shortcut.toKeyEquivalent() {
+			content()
+					.keyboardShortcut(keyEquivalent, modifiers: shortcut.toEventModifiers())
```

Comment:
Would using `keyboardShortcut` here cause it to both trigger the Button in `content()` and also the KeyboardShortcuts event listener.

Reply:
I did test for this - if I have the app active, as well as a "global" shortcut handler installed; I only get one event. I just presumed KeyboardShortcuts was being somehow smart